### PR TITLE
[stable/datadog][v2.0.1] improve logs fields

### PR DESCRIPTION
**Pull request by clamoriniere**
fix several issues introduced with v2.0
* bad cluster-agent deployment pod template indentation in `volumesMount` part.
* add missing logs parameters refactoring.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/datadog/templates/container-agent.yaml [hunk 1]
```diff
@@ -81,13 +81,13 @@
     - name: DD_KUBERNETES_POD_ANNOTATIONS_AS_TAGS
       value: '{{ toJson .Values.datadog.podAnnotationsAsTags }}'
     {{- end }}
-    {{- if .Values.datadog.logsEnabled }}
+    {{- if or .Values.datadog.logs.enabled .Values.datadog.logsEnabled }}
```
Comment (xornivore) [inline stable/datadog/templates/container-agent.yaml]: This change implies that the default for `DD_LOGS_ENABLED=false` will never change, is that safe to assume?

File: stable/datadog/templates/container-agent.yaml [hunk 2]
```diff
@@ -81,13 +81,13 @@
...
     - name: DD_LOGS_ENABLED
-      value: {{.Values.datadog.logsEnabled | quote}}
+      value: "true"
     {{- end }}
-    {{- if .Values.datadog.logsConfigContainerCollectAll }}
+    {{- if or .Values.datadog.logs.enabled .Values.datadog.logsConfigContainerCollectAll }}
```
Comment (vboulineau) [inline stable/datadog/templates/container-agent.yaml]:
```suggestion
    {{- if or .Values.datadog.logs.containerCollectAll .Values.datadog.logsConfigContainerCollectAll }}
```

**Review by vboulineau — CHANGES_REQUESTED** [review]

File: stable/datadog/templates/container-agent.yaml (hunk 1 above)
Comment (clamoriniere) [reply stable/datadog/templates/container-agent.yaml]: right do you think if I do
```suggestion
    - name: DD_LOGS_ENABLED
      value: {{  (default false (or .Values.datadog.logs.enabled .Values.datadog.logsEnabled)) | quote}}
```

Comment (clamoriniere) [reply stable/datadog/templates/container-agent.yaml]: right, what do you think if I do:

```suggestion
    - name: DD_LOGS_ENABLED
      value: {{  (default false (or .Values.datadog.logs.enabled .Values.datadog.logsEnabled)) | quote}}
```

Comment (clamoriniere) [reply stable/datadog/templates/container-agent.yaml]: right, what do you think if I do:

```suggestion
    - name: DD_LOGS_ENABLED
      value: {{  (default false (or .Values.datadog.logs.enabled .Values.datadog.logsEnabled)) | quote}}
```

Comment (xornivore) [reply stable/datadog/templates/container-agent.yaml]: Sure this works!

**Review by xornivore — APPROVED** [review]
👍

**Review by mbakht — COMMENTED** [review]
👍

**Comment by madsonic** [comment]
typo in migration guide. it should be under `datadog` block `.datadog.logs.enabled`

https://github.com/helm/charts/pull/21086/files#diff-d5634f0483d5b5c34d9d6fd00ca95c6fR69

**Pull request merged**